src/common/chains.ts

```diff
@@ -9,3 +11,383 @@ export enum SupportedChainId {
+ */
+export type ChainId = number
+
+export interface WebUrl {
+  url: string
+  name: string
+}
+
+export type ChainRpcUrls = {
+  http: readonly string[]
+  webSocket?: readonly string[] | undefined
+}
+
+export type ChainContract = {
+  address: string
+  blockCreated?: number | undefined
+}
+
+export type ChainContracts = {
+  multicall3?: ChainContract
+  ensRegistry?: ChainContract
+  ensUniversalResolver?: ChainContract
+  universalSignatureVerifier?: ChainContract
+}
+
+/**
+ * A chain on the network.
+ *
+ * Probably we could use the viem chain definition, I think multicall3, and ensRegistry and the types defined there can be handy. But for now we are using a simplified version.
+ *
+ * For a list of chains, see: https://github.com/wevm/viem/tree/main/src/chains/definitions
+ */
+export interface ChainInfo {
+  /**
+   * The chain id.
+   */
+  readonly id: ChainId
+
+  /**
+   * Name of the chain as defined in
+   */
+  readonly name: string
+
+  /**
+   * Label of the chain. Field used for display purposes.
+   */
+  readonly label: string
+
+  /**
+   * Native currency of the chain.
+   */
+  readonly nativeCurrency: TokenInfo
+
+  /**
+   * ERC-3770 address prefix
+   *
+   * See https://eips.ethereum.org/EIPS/eip-3770
+   */
+  readonly addressPrefix: string
+
+  /**
+   * Whether the chain is a testnet.
+   */
+  readonly isTestnet: boolean
+
+  /**
+   * Contracts of the chain.
+   */
+  readonly contracts: ChainContracts
+
+  /**
+   * Main color of the chain, used for presentation purposes.
+   */
+  readonly color: string
+
+  /**
+   * Logo of the chain.
+   */
+  readonly logo: { light: string; dark: string }
+
+  /**
+   * Documentation of the chain.
+   */
+  readonly docs: WebUrl
+
+  /**
+   * Website of the chain.
+   */
+  readonly website: WebUrl
+
+  /**
+   * Block explorer of the chain.
+   */
+  readonly blockExplorer: WebUrl
+
+  /**
+   * Bridges of the chain.
+   */
+  readonly bridges?: WebUrl[]
+
+  /**
+   * RPC URLs of the chain.
+   */
+  readonly rpcUrls: {
+    [key: string]: ChainRpcUrls
+    default: ChainRpcUrls
+  }
+}
+
+const CHAIN_ASSETS = 'https://raw.githubusercontent.com/cowprotocol/cowswap/refs/heads/develop/libs/assets/src/cow-swap'
+export const EthereumLogo = `${CHAIN_ASSETS}/network-mainnet-logo.svg`
+const GnosisChainLogo = `${CHAIN_ASSETS}/network-gnosis-chain-logo.svg`
+const ArbitrumOneLogoLight = `${CHAIN_ASSETS}/network-arbitrum-one-logo-blue.svg`
+const ArbitrumOneLogoDark = `${CHAIN_ASSETS}/network-arbitrum-one-logo-white.svg`
+const BaseLogo = `${CHAIN_ASSETS}/network-base-logo.svg`
+const SepoliaLogo = `${CHAIN_ASSETS}/network-sepolia-logo.svg`
+/**
+ * Mainnet chain info.
+ *
+ * See also https://github.com/wevm/viem/blob/main/src/chains/definitions/mainnet.ts
+ */
+export const mainnet: ChainInfo = {
+  id: SupportedChainId.MAINNET,
+  name: 'mainnet',
+  label: 'Ethereum',
+  nativeCurrency: WRAPPED_NATIVE_CURRENCIES[SupportedChainId.MAINNET],
```

Comment:
Is there any reason for using WETH as native currency?
Should we maybe add another property for the wrapped token?

Reply:
Actually no. Fixing it everywhere....